### DISCUSSION TITLE
How would I run addon.settings.get in a popup

### DISCUSSION THREAD

**Discussion by AwesomeKalin** [ScratchAddons/ScratchAddons#1658] [Help]
I'm currently working on a change that requires addon.settings.get to be run, but how would I do this in a popup?

**Comment by retronbv** [comment]
I don’t think you can.
Edit: because the popup is not an addon, it doesn’t get the addon apis passed to it. I hope that makes sense lol

**Reply by AwesomeKalin** [reply]
Maybe you can if you import a persistent script?

**Reply by retronbv** [reply]
> Maybe you can if you import a persistent script?

What addon would this be for?

**Reply by AwesomeKalin** [reply]
Cloud Games. I'm writing a change that would allow custom studios

**Reply by retronbv** [reply]
Oh, I’m dumb.
Honestly I don’t see how using a persistent script would fix it. Even if we use a PS not all users have this addon enabled. (I forget has It been implemented where you can turn off tabs in the popup?’

**Reply by AwesomeKalin** [reply]
You can

**Reply by retronbv** [reply]
You can what? Turn off tabs in the popup or somehow pass the addon apis to a popup page?

**Reply by AwesomeKalin** [reply]
If you turn of the cloud games add-on, it removes the tab

**Reply by retronbv** [reply]
Thank you for clarifying. Reading the documentation on PS I don’t think you can pass the addon apis to the popup. Yes, you can get the addon apis in PS.

**Comment by cobaltt7** [comment]
Run this line:
```js
chrome.storage.sync.get("addonSettings", s => window.settings = s.addonSettings['cloud-games'])
```
The variable `settings` is now an associative array of all of the settings for the `cloud-games` addon.

Hope this helped!

**Reply by WorldLanguages** [reply]
Nice hack. Keep in mind reading storage is async so settings won't be available immediately.

**Reply by AwesomeKalin** [reply]
How long would it take for it to be avaliable

**Reply by TheColaber** [reply]
less than a second I suppose.

**Reply by WorldLanguages** [reply]
@AwesomeKalin You should provide a function as a callback, instead of guessing how much it could take.

**Comment by WorldLanguages** [comment]
As of now popups are pretty "underpowered" you could say, see #1234. I'd suggest you join our development Discord server so that we can discuss the best workaround for your addon?

**Reply by cobaltt7** [reply]
I thought the discord was only for the team members?

**Reply by retronbv** [reply]
they are making an addon? why arent they considered "on the team"

**Reply by WorldLanguages** [reply]
@RedGuy12 No? Feel free to join. Being a team member doesn't actually mean much lol

**Reply by AwesomeKalin** [reply]
What is the link to the discord?

**Reply by retronbv** [reply]
It’s in the readme
